src/content/developers/docs/data-availability.md

```diff
@@ -0,0 +1,153 @@
+
+Each block has two major parts: 
+
+- The **block header**: This contains general information (metadata) about the block, such as the timestamp, block hash, block number, etc.
+- The  **block body**: This contains the actual transactions processed as part of the block. 
+
+[Miner nodes](/developers/docs/consensus-mechanisms/pow/mining/) (or validators in [proof-of-stake](/developers/docs/consensus-mechanisms/pos/)) proposing new blocks must publish both the block header and block body. This allows nodes participating in [consensus](/developers/docs/consensus-mechanisms/) (full nodes) to download the block and confirm that transactions follow Ethereum’s consensus protocol. 
+
+[Full nodes](/developers/docs/nodes-and-clients/#full-node) are different from [light nodes](/developers/docs/nodes-and-clients/#light-node) that only download the block header. Without full nodes verifying transactions, block proposers could get away with inserting malicious transactions in blocks. 
+
+### The data availability problem {#the-data-availability-problem}
+
+The data availability problem is encapsulated in one question: “How do we verify that the data for a newly produced block is actually available?” As mentioned, the security Ethereum rests on the assumption that full nodes have access to block data. 
+
+If a block producer publishes a new block, but hides some (or all) of the underlying information, they might be able to finalize invalid transactions. Even if a block is valid, withholding data still has negative implications for users and the functionality of the network.   
+
+Most early discussions around data availability concerned the visibility of transaction data to full nodes (validators) participating in Ethereum’s consensus. However, newer DA research is also focused on verifying data availability with light clients. For light clients, the problem of data availability concerns validating a block’s data availability without having to download the entire block. 
+
+### Data availability vs. data retrievability {#data-availability-vs-data-retrievability}
+
+Data availability is a different idea from data retrievability. Data availability is the ability of peer-to-peer nodes to download transaction data for a block while it is being proposed for addition to the chain. In other words, data availability is relevant when a block is yet to pass consensus. 
+
+Data retrievability is the ability of nodes to download transactions from a block *after* it has passed consensus. Essentially, data retrievability is concerned with retrieving data from historical blocks. A block becomes “historical” when other blocks have been built on it. 
+
+The core Ethereum protocol is primarily concerned with data availability, not data retrievability. It isn’t the blockckhain’s duty to store data for every transaction it reaches consensus on. Besides, storing state data in perpetuity increases hardware requirements for full nodes and may harm decentralization. 
```

Comment:
> This seems a little simplistic. Obviously, Ethereum needs to allow data to be retrieved at some point past the block being confirmed.

Added a more specific description of data retrievability

Reply:
> > storing state data in perpetuity increases hardware requirements for full node
> 
> This technically isn't true. Full nodes are typically pruned, and requests are made to archive nodes to reconstruct the entirety of the chain.

Here are some Vitalik quotes for context on why I included that part:

> One of the longest and still unresolved challenges in the Ethereum protocol is that of how to manage the problem of growing state. Many actions in Ethereum (creating an account, filling a storage slot, sending ETH to a new account…) add objects to the state that all full nodes must store in order to be able to verify or produce new blocks. These actions require a one-time gas cost to be paid by the sender of the transaction, but impose permanent ongoing costs on the network, and the nodes that need to store this new data (and future nodes that need to download it when syncing).

> This is a major imbalance in the system’s design, and risks the Ethereum system becoming more and more difficult to use over time as the state fills with no-longer-needed “junk data”. The purpose of this document is to describe in more detail what the problem is, and what are some of the paths toward solving it. If we successfully implement a solution, it could pave the way to safely making significant increases to the block gas limit.

> The Ethereum state size is growing quickly. It is currently around 35 GB for just the state and over 100 GB including all Merkle proofs, and is increasing by roughly half that amount per year. State storage is additionally a weak point in Ethereum’s economics: it is the only mechanism by which a participant can pay a cost once to burden consensus nodes forever. In order to maintain the scalability and sustainability of Ethereum, we need some solution. Two paths to a solution exist, and have existed for a long time: weak statelessness and state expiry: